### PR TITLE
Enhance Synthetic Data Generators for Payment Amounts with Randomized and Statistical Distributions

### PR DESCRIPTION
**Pull request by nitay1234-mcpp**
This pull request enhances the synthetic data generators used in the test_payment_processing.py file by introducing randomized payment amounts based on a log-normal distribution to better simulate real-world payment behavior. It also adds a randomized card information generator to diversify test scenarios.

These enhancements improve test coverage and realism of payment processing tests, helping to uncover edge cases and performance issues more effectively.

Existing test cases remain unchanged to maintain current validation while benefiting from added randomized testing.

### CODE REVIEW

**Comment by nitay1234-mcpp** [comment]
Regarding the flaky tests in this PR, I see that the new test "test_payment_processing_randomized" is marked with a retry decorator to handle intermittent failures, which is a good approach to mitigate flakiness. This aligns with the regression issue I mentioned back in March. The enhancements in synchronization and retry logic for webhook handling and multiple payment attempts also appear to improve stability. Overall, this addresses the flaky test concerns effectively.

**Review by nitay1234-mcpp — COMMENTED** [review]
The enhancements to the synthetic data generators in this PR are a valuable addition. Using a log-normal distribution for generating payment amounts realistically simulates the variability seen in real-world payment data, which is a significant improvement over fixed or simplistic values. The clamping of amounts to a sensible range ensures edge cases are handled appropriately without causing test failures due to extreme values.

The randomized card information generator adds diversity to test scenarios by including valid, fraudulent, and not accepted card numbers, improving the robustness of the payment processing tests.

The new test method that exercises these generators across multiple iterations with assertions on status and performance enhances the overall test coverage and helps identify edge cases and performance issues more effectively.

Overall, these enhancements improve the realism and coverage of payment processing tests and are well implemented.

**Review by nitay1234-mcpp — COMMENTED** [review]
I appreciate the enhancements made to the synthetic data generators for payment amounts. The introduction of randomized payment amounts based on a log-normal distribution adds valuable realism to the test scenarios. This kind of randomized testing is crucial for identifying edge cases and uncovering potential performance issues that deterministic tests may miss. These improvements will undoubtedly strengthen the robustness and coverage of the payment processing tests. Great work!

**Comment by nitay1234-mcpp** [comment]
This is the regression I told you about in March. Please review the existing comment for context regarding flaky tests and retry logic improvements.

**Review by nitay1234-mcpp — COMMENTED** [review]
This pull request enhances synthetic data generators for payment amounts, which is highly relevant to the regression suite as it improves test coverage and realism. The randomized and statistically distributed payment amounts will help uncover edge cases and potential issues more effectively during regression testing.

**Comment by nitay1234-mcpp** [comment]
The test coverage improvements in this PR are excellent, especially with the addition of randomized payment processing tests. Generating payment amounts based on a log-normal distribution and introducing randomized card information add valuable realism and diversity to the test scenarios. This approach enhances the ability to uncover edge cases and potential performance issues that fixed-value tests might miss. The use of flaky reruns also helps maintain stability despite the randomness. Overall, these enhancements significantly boost the robustness and reliability of the payment processing tests. Great work!

**Comment by nitay1234-mcpp** [comment]
Please note the regression test I warned about in March. Ensure the enhancements in this PR do not cause any regressions or failures in that test. Let me know if you need any clarifications or assistance with the regression suite.

File: test_payment_processing.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import pytest
```
Comment (nitay1234-mcpp) [inline test_payment_processing.py:1]: For clarity, it would be beneficial to explicitly state the acceptance criteria for the synthetic data generators in the test documentation or comments. This should include the expected range of generated payment amounts, the distributions used, and the types of card information randomized. Clear acceptance criteria help reviewers and future maintainers understand the goals and boundaries of these enhancements.

**Review by nitay1234-mcpp — COMMENTED** [review]
Added a comment suggesting to explicitly state the acceptance criteria for the synthetic data generators in the test documentation or comments for better clarity and maintainability.
